python-checks/src/test/resources/checks/regex/possessiveQuantifierContinuationCheck.py

```diff
@@ -1,7 +1,27 @@
 import re
 
-match1 = re.match(r"a++abc", "aaaabc", re.DOTALL) # Noncompliant {{Change this impossible to match sub-pattern that conflicts with the previous possessive quantifier.}}
-match2 = re.match(r"\d*+[02468]", "1234", re.DOTALL) # Noncompliant {{Change this impossible to match sub-pattern that conflicts with the previous possessive quantifier.}}
+def foo():
+    match1 = re.match(r"a++abc", "aaaabc", re.DOTALL) # Noncompliant {{Change this impossible to match sub-pattern that conflicts with the previous possessive quantifier.}}
+    match2 = re.match(r"\d*+[02468]", "1234", re.DOTALL) # Noncompliant {{Change this impossible to match sub-pattern that conflicts with the previous possessive quantifier.}}
 
-match3 = re.match(r"aa++bc", "aaaabc", re.DOTALL) # Compliant, for example it can match "aaaabc"
-match4 = re.match(r"\d*+(?<=[02468])", "1234", re.DOTALL) # Compliant, for example, it can match an even number like "1234"
+    match3 = re.match(r"aa++bc", "aaaabc", re.DOTALL) # Compliant, for example it can match "aaaabc"
+    match4 = re.match(r"\d*+(?<=[02468])", "1234", re.DOTALL) # Compliant, for example, it can match an even number like "1234"
+
+    re.match(r"a+abc", "aaaabc", re.DOTALL)
```

Comment:
Do we need to have this `re.DOTALL` flag in all tests (as well as in the rule description examples)?
In my opinion, if we don't need it, we shouldn't have it (at least not everywhere), as it adds a bit of noise.

I don't think it matters much and we can keep it like this, but this is more of a general point that we should try to keep examples as minimal as possible.